src/containers/Forms/Payment/CreatePaymentForm.js

```diff
@@ -23,20 +19,22 @@ const PaymentForm = ({ isOpen, handleClose, studId, title }) => {
     amount: '',
   });
 
+  const [open, setOpen] = useState(false);
+
   useEffect(() => {
     dispatch(fetchStudents());
   }, [dispatch]);
 
-  const inputChangeHandler = (e) => {
-    const { name, value } = e.target;
+  useEffect(() => {
+    isOpen ? setOpen(true) : setOpen(false);
+  }, [isOpen]);
 
-    studId
-      ? setPayment({ ...payment, studentId: studId, [name]: value })
-      : setPayment({
-          ...payment,
-          date: e.target.name === 'date' ? e.target.value : payment.date,
-          amount: e.target.name === 'amount' ? e.target.value : payment.amount,
-        });
+  const inputChangeHandler = (e) => {
+    setPayment({
+      ...payment,
+      date: e.target.name === 'date' ? e.target.value : payment.date,
+      amount: e.target.name === 'amount' ? e.target.value : payment.amount,
+    });
```

Comment:
Можно так сократить, и если изменяешь стейт с использованием текущего стейта лучше оборачивать в функцию, тогда стейт гарантированно будет актуальным
```suggestion
  const inputChangeHandler = (e) => {
  const { name, value } = e.target;
    setPayment((state) => ({
      ...state,
      [name]: value,
    }));
  };
```

Reply:
исправлено